api/src/main/kotlin/org/jellyfin/apiclient/api/client/KtorClient.kt

```diff
@@ -84,34 +106,13 @@ open class KtorClient(
 		queryParameters: Map<String, Any?> = emptyMap(),
 		body: Any? = null
 	): Response<T> {
-		// Check if the base url is not null
-		val baseUrl = checkNotNull(baseUrl)
-
 		val response = client.request<HttpResponse> {
 			this.method = method
-
-			url {
-				// Create from base URL
-				takeFrom(baseUrl)
-
-				// Replace path variables
-				val path = createPath(pathTemplate, pathParameters)
-				// Assign path making sure to remove duplicated slashes between the base and appended path
-				encodedPath = "${encodedPath.trimEnd('/')}/${path.trimStart('/')}"
-
-				// Append query parameters
-				queryParameters
-					.filterNot { it.value == null }
-					.forEach {
-						parameters.append(it.key, it.value.toString())
-					}
-			}
-
+			url(createUrl(pathTemplate, pathParameters, queryParameters))
 			header(HttpHeaders.Authorization, createAuthorizationHeader())
 
-			if (body != null) {
+			if (body != null)
 				this.body = defaultSerializer().write(body)
```

Comment:
You should maybe rename the passed function argument to decrease possible confusion..